### ISSUE TITLE
Password Gorilla Mac OSX (15372) damaged and cannot open

### ISSUE THREAD

**Issue by KillerSquid** [zdia/gorilla#130]
I downloaded your program because it's the one our Windows machines use at work for password management but everytime I try to open it I get the following:

"Password Gorilla" is damaged and can't be opened.  You should move it to the Trash.

I set the application to run in 32 bit mode and am getting the same message.  Any thoughts?
Thanks!

**Comment by newgoliath** [comment]
I had to download it with `curl -o gorilla.zip http://gorilla.dp100.com/downloads/gorilla.mac.15373.zip`  and unzip it.  Otherwise the file was corrupted.

Yosemite.

**Comment by rich123** [comment]
@newgoliath
No idea what went wrong.  The files at that URL are just static files on a filesystem served by Apache running on Linux.  Nothing special, and nothing more than plain static file serving.

I show twenty-four downloads of that .zip file in my Apache logs for today (March 2), all twenty-four report a 200 success in the log file.  As far as Apache was concerned, everything downloaded properly.

**Comment by acidjunk** [comment]
The OSX version from the .app shows a message that's it's corrupted when you try to launch it with OSX gatekeeper enabled. @rich123: I could create a officially signed version if you want it. (probably need some guidance about how to build it)

**Comment by newgoliath** [comment]
I didn't need to reonfig with gatekeeper when using `curl` and `unzip`.

**Comment by rich123** [comment]
@acidjunk: Creating an officially signed 'Apple' version would be good, hopefully that would alleviate the gatekeeper "it is damaged" (I nominate that for 'worst error message wording' for this years trials) message.  How does one go about "signing" the package?

**Comment by rich123** [comment]
@newgoliath: That suggests to me that Apple's "it is damaged" system works much like the similar misfeature that win7 contains.  The browser, upon downloading something from the internet, adds an extra bit of metadata in an alternate file stream to the file that says "this came from the internet".  Win7's "it must be bad for you" subsystem uses that extra data to pop up a "this came from the 'net, do you really want to run it".  It sounds like Apple's system works similarly, and Apple forgot to have curl tag the download as "scary".

**Comment by leeschumacher** [comment]
I encountered a similar issue (app is damaged popup on start), but it wasn't gatekeeper.  Somehow the installation had the owner set to id/group as 501:wheel, and the permissions on all the files/directories in the app were set to user rwx only (eg, 0700).  All the files/directories inside were set the same way.  I had to go in with 'sudo chmod a+r' on all the files, and a+x on the directories in order to get it to run.

**Comment by rich123** [comment]
Which specific package/file did you get this error from?  This is good to know, because fixing this specific issue will be easy.  Just have to setup owner/permissions properly for the package bundle.

**Comment by leeschumacher** [comment]
The version i have is 1.5.3.7.  I'm pretty sure it was downloaded from gorilla.mac.15373.zip (the md5 for gorilla.aqua from that zip matches the version in my /Applications directory).

Re-downloading and unzipping it results in set of files with the correct permissions, so perhaps it was operator error on my part that resulted in the problem.  I'm not sure how that would have happened by accident.

**Comment by rich123** [comment]
No, during the latency between my reply and yours, I checked the zip on the download server, and it had the permissions you described encoded into it.  I changed the permissions to be more permissive, repackaged the zip, and put up a new gpg .sig file matching the newly packaged zip.

I did not change the version number because there was no change to the Gorilla code in any way, all that changed was the permissions in the wrapper used by MacOS to convert it into a Mac application bundle.

**Comment by leeschumacher** [comment]
Damn it.  Gaslit again ... :)

**Comment by LouMishew** [comment]
May be I'm late on this, but I'll share the solution anyway:   OS X actually does add a metadata bit, named com.apple.quarantine, which prevents the file to be opened by "ordinary" customers. To clear that bit the command xattr can be used:
xattr -c <downloaded_gorilla.zip_file>

**Comment by rich123** [comment]
Yes, that is the way to remove the gatekeeper "mark" on the downloaded file and allow it to run.

It can be found here on github in the MacOS Gatekeeper page: https://github.com/zdia/gorilla/wiki/MacOS-Gatekeeper

And as well, it can also be found in the "MacOS Gatekeeper Information" page from the download page: https://gorilla.dp100.com/downloads/MacOS-Gatekeeper.html

**Comment by Rolice** [comment]
At last I have managed to run it. The important point is indeed documented [here](https://github.com/zdia/gorilla/wiki/MacOS-Gatekeeper) but it was not obvious for me (to execute): `xattr -d com.apple.quarantine Password\ Gorilla.app/`

**Comment by rich123** [comment]
I changed the wiki page to say "execute" instead of "run" for step #4 in the set of steps to turn off the quarantine.  If you have any other suggestions for how to make this page easier to understand I am open to suggestions.
